### ISSUE TITLE
WebSocket issue

### ISSUE THREAD

**Issue by uBlock-user** [gorhill/uBlock#2300]
### Describe the issue

Solidfiles.com makes a websocket connection request and uBO doesn't block it even when `||$websocket` OR `||$websocket^$important` OR `||$websocket,domain=solidfiles.com` filter is present.

### One or more specific URLs where the issue occurs

http://www.solidfiles.com/v/aWwVXwNGDD8L7

### Screenshot in which the issue can be seen

https://i.gyazo.com/7a823cc20da6853f56b8becc8372dcfb.png

### Steps for anyone to reproduce the issue

1) Add `||$websocket` OR `||$websocket^$important` OR `||$websocket,domain=solidfiles.com` filter to `My Filters` and open the logger.

2) visit http://www.solidfiles.com/v/aWwVXwNGDD8L7

3) Notice uBO sees the websocket connection request in the logger, however it remains unblocked despite having the blocking filter.
 

### Your settings

- Browser/version: Chromium 57
- uBlock Origin version: 1.10.4

### Notes

This warning appears in the console - https://i.gyazo.com/f95a0e35658dbbe895d1911660ba86af.png

**Comment by uBlock-user** [comment]
Is it getting blocked at your end @gorhill ?

**Comment by gorhill** [comment]
Yes, and the content policy security violation is reported in the console.

**Comment by uBlock-user** [comment]
Not on my end, dynamic rules work, it seems.

**Comment by gorhill** [comment]
Most likely is that you are overlooking something.

> dynamic rules work

What does that mean? Specifics please.

**Comment by uBlock-user** [comment]
> What does that mean? Specifics please.

adding `* ws://www.solidfiles.com websocket block` in `My Rules` blocks the request.

**Comment by gorhill** [comment]
There could be an exception filter somewhere preventing your filter to work. As of now, I observe that exception for `*$websocket`-like filters are not reported in the logger.

**Comment by uBlock-user** [comment]
> There could be an exception filter somewhere preventing your filter from being applied

Even when using `||$websocket^$important` ?

This happens on Chromium 55 with default settings

https://i.gyazo.com/b8eaee04a1aeab0925c26e3f35544bb0.png

**Comment by gorhill** [comment]
> Even when using `||$websocket^$important` ?

No, not when using `important`. I just can't reproduce, so you will have to investigate on your side.

**Comment by uBlock-user** [comment]
I mean I'm using `||$websocket^$important` and it still happens.

**Comment by uBlock-user** [comment]
I've already investigated and I don't know what more I can do.

Same thing happens on `github.com`. I already have `||websocket` and yet github is able to open websocket connection - https://i.gyazo.com/de63083aadd549237a1416428b1f1f4f.png

**Comment by uBlock-user** [comment]
I don't have anything else to go any further, so I will close it as this is an impasse.

**Comment by gorhill** [comment]
If nobody else can reproduce, you have something on your side you are overlooking. Are you using another extension which also tries to modify the CSP header?

I can't reproduce = I can't investigate.

**Comment by uBlock-user** [comment]
I run 3 extensions in total - uBO, uBO Extra and uMatrix. Could uMatrix be causing this ? I disabled uMX and I was still able to reproduce the issue.

**Comment by gorhill** [comment]
Is _My filters_ enabled?

If I could reproduce, I would put a breakpoint at [that line](https://github.com/gorhill/uBlock/blob/d913eea1f66451423e328101b10ee0c3e8611689/src/js/traffic.js#L472) to find out the value of `websocketResult`.

**Comment by uBlock-user** [comment]
Yes, it is and it's empty too.

**Comment by gorhill** [comment]
What count of filters does _3rd-party filters_ reports for _My filters_?

**Comment by uBlock-user** [comment]
`0 used out of 0`

**Comment by gorhill** [comment]
Can you cut & paste here *verbatim* the content of _My filters_?

**Comment by uBlock-user** [comment]
There is nothing to cut and paste, it's empty, here's the pic - https://i.gyazo.com/19def51df24b98b20981a6f2cc254ccb.png

**Comment by gorhill** [comment]
> There is nothing to cut and paste, it's empty

You are losing me. How is uBO supposed to block websocket if you do not have the filter you said you have in your opening comment?

**Comment by uBlock-user** [comment]
> You are losing me. How is uBO supposed to block websocket if you do not have the filter you said you have in your opening comment?

my apologies, I was testing with all the 3 filters one by one and forgot to add back the filter, I use to blanket block all websocket connections, after all the tests failed.

Here's how it looks normally - https://i.gyazo.com/2e06e42d8b0debaa5e31c6f6cdf49e48.png

**Comment by gorhill** [comment]
So now what is _My filters_ count?

**Comment by uBlock-user** [comment]
Thank you. It works now.

`My Filters` count is `1 out of 1` and always is, but for some unknown reason it wasn't being applied on `github.com` and `solidfiles.com` back when I posted about them. Now I removed it and re-added it in `My Filters` tab and tested again and is functioning normally.

`My Filters` - https://i.gyazo.com/43acc30e167f039b95eb46d538953c4d.png

`Console` - https://i.gyazo.com/9c9388ef7499b24d0c5790af2c55eba8.png

Sorry for the trouble.
